radixdlt/src/main/java/com/radixdlt/middleware2/network/MessageCentralSyncCommittedNetwork.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.middleware2.network;
+
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.consensus.sync.StateSyncNetwork;
+import com.radixdlt.consensus.sync.SyncRequest;
+import com.radixdlt.middleware2.CommittedAtom;
+import com.radixdlt.network.addressbook.Peer;
+import com.radixdlt.network.messaging.MessageCentral;
+import com.radixdlt.network.messaging.MessageListener;
+import com.radixdlt.universe.Universe;
+import io.reactivex.rxjava3.core.Observable;
+import java.util.List;
+import java.util.Objects;
+import javax.inject.Inject;
+
+/**
+ * Network interface for syncing committed state using the MessageCentral
+ */
+public final class MessageCentralSyncCommittedNetwork implements StateSyncNetwork {
+	private final int magic;
+	private final MessageCentral messageCentral;
+
+	@Inject
+	public MessageCentralSyncCommittedNetwork(
+		Universe universe,
+		MessageCentral messageCentral
+	) {
+		this.magic = universe.getMagic();
```

Comment:
Can we pass in magic directly? Or will we need universe for something?

Reply:
no dont need it for anything. Should actually just aim to remove it as I believe `magic` isn't protected by any signature.

Reply:
All of the elements used to compute the magic are signed by the universe creation key — they are fields in the universe, and the universe is signed.